library/ceph_crush.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+
+#
+# Copyright (c) 2018 Red Hat, Inc.
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: ceph_crush
+
+author: Sebastien Han <seb@redhat.com>
+
+short_description: Create Ceph CRUSH hierarchy
+
+version_added: "2.6"
+
+description:
+    - By using the hostvar variable 'osd_crush_location'
+    ceph_crush creates buckets and places them in the right CRUSH hierarchy
+
+options:
+    cluster:
+        description:
+            - The ceph cluster name.
+        required: false
+        default: ceph
+    location:
+        description:
+            - osd_crush_location dict from the inventory file. It contains
+            the placement of each host in the CRUSH map.
+        required: true
+    containerized:
+        description:
+            - Weither or not this is a containerized cluster. The value is
+            assigned or not depending on how the playbook runs.
+        required: false
+        default: None
+'''
+
+EXAMPLES = '''
+- name: configure crush hierarchy
+  ceph_crush:
+    cluster: "{{ cluster }}"
+    location: "{{ hostvars[item]['osd_crush_location'] }}"
+    containerized: "{{ docker_exec_cmd }}"
+  with_items: "{{ groups[osd_group_name] }}"
+  when:
+    - crush_rule_config
+'''
+
+RETURN = '''#  '''
+
+from ansible.module_utils.basic import AnsibleModule
+import datetime
+
+
+def fatal(message, module):
+    '''
+    Report a fatal error and exit
+    '''
+    if module:
+        module.fail_json(msg=message, rc=1)
+    else:
+        raise(Exception(message))
+
+
+def generate_cmd(cluster, subcommand, bucket, bucket_type, containerized=None):
+    '''
+    Generate command line to execute
+    '''
+    cmd = [
+        'ceph',
+        '--cluster',
+        cluster,
+        'osd',
+        'crush',
+        subcommand,
+        bucket,
+        bucket_type,
+    ]
+    if containerized:
+        for piece in reversed(containerized.split()):
```

Comment:
I looks like you're trying to prepend the the docker exec command here. This might be a cleaner way to achieve the same result:

```
cmd = containerized.split() + cmd
```

Reply:
Thanks!